Mathlib/Data/FunLike/Basic.lean

```diff
@@ -125,6 +125,10 @@ injective coercion to functions from `α` to `β`.
 
 This typeclass is used in the definition of the homomorphism typeclasses,
 such as `ZeroHomClass`, `MulHomClass`, `MonoidHomClass`, ....
+
+Lean will try to synthesize non-dependent `FunLike` typeclasses first by default. Use the command
+`attribute [local instance high] FunLike.hasCoeToFun` if you want dependent `FunLike` typeclasses
+to be synthesized.
```

Comment:
Could you explain in the comment why the dependent version isn't just a fallback for the non-dependent version? (I assume it's because of `outParam`s.)

Reply:
I think this should really just be giving performance advice, https://github.com/leanprover-community/mathlib4/pull/7905#discussion_r1372943100